src/pages/Login/Login.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+
+import { useAppDispatch } from 'store/hooks';
+import { setUser, UserState } from 'store/UserSlice';
+
+interface Fields {
+  login: string;
+  password: string;
+}
+
+const Login = () => {
+  const { register, handleSubmit } = useForm<Fields>();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+
+  const verifyUser = (data: Fields): Promise<UserState> => {
+    return new Promise((resolve, reject) => {
+      if (data.login === 'admin' && data.password === '12345') {
+        resolve({ firstName: 'Mykola', lastName: 'Gumeniuk', isAuth: true });
+      }
+      reject('kto ty voin?');
+    });
+  };
```

Comment:
Even if it's a mock function, move it outside of the component. By doing that you will increase performance